### PR TITLE
cloudapi/v2: Use manifest-id-only job

### PR DESCRIPTION
**Pull request by Gundersanne**
job dependencies:
depsolve -> manifest -> osbuild

This allows the compose handler to return the osbuild job id
immediately.

### CODE REVIEW

**Review by teg — APPROVED** [review]
This looks great to me!

File: internal/cloudapi/v2/v2.go [hunk 1]
```diff
@@ -397,22 +360,110 @@ func (h *apiHandlers) PostCompose(ctx echo.Context) error {
...
+		Targets:  []*target.Target{irTarget},
+		Exports:  imageType.Exports(),
+	}, manifestJobID)
 	if err != nil {
 		return HTTPErrorWithInternal(ErrorEnqueueingJob, err)
 	}
 
 	ctx.Logger().Infof("Job ID %s enqueued for operationID %s", id, ctx.Get("operationID"))
 
+	// start 1 go routing which requests datajob type
```
Comment (ondrejbudai) [inline internal/cloudapi/v2/v2.go:384]: goroutine

File: internal/cloudapi/v2/v2.go [hunk 2]
```diff
@@ -397,22 +360,110 @@ func (h *apiHandlers) PostCompose(ctx echo.Context) error {
...
-		Targets:  []*target.Target{imageRequest.target},
-		Exports:  imageRequest.exports,
-	})
+	manifestJobID, err := h.server.workers.EnqueueManifestJobByID(&worker.ManifestJobByID{}, depsolveJobID)
+	if err != nil {
+		return HTTPErrorWithInternal(ErrorEnqueueingJob, err)
+	}
+
+	id, err := h.server.workers.EnqueueOSBuildAsDependency(arch.Name(), &worker.OSBuildJob{
+		Manifest: nil, // distro.Manifest([]byte{}),
```
Comment (ondrejbudai) [inline internal/cloudapi/v2/v2.go:374]: stray comment?

File: internal/cloudapi/v2/v2.go [hunk 3]
```diff
@@ -397,22 +360,110 @@ func (h *apiHandlers) PostCompose(ctx echo.Context) error {
...
+				continue
+			}
+			if err != nil {
+				logrus.Errorf("Error requesting manifest job: %v", err)
+				return
+			}
+			break
+		}
+
+		finishJob := func(jr worker.ManifestJobByIDResult) {
```
Comment (ondrejbudai) [inline internal/cloudapi/v2/v2.go:404]: I think I would rather use `defer`. See how it's done elsewhere: https://github.com/osbuild/osbuild-composer/blob/main/cmd/osbuild-worker/jobimpl-osbuild.go#L64

File: internal/cloudapi/v2/v2.go [hunk 4]
```diff
@@ -397,22 +360,110 @@ func (h *apiHandlers) PostCompose(ctx echo.Context) error {
...
+	go func(workers *worker.Server, manifestJobID uuid.UUID, b *blueprint.Customizations, options distro.ImageOptions, repos []rpmmd.RepoConfig, seed int64) {
+		// wait until job is in a pending state
+		var token uuid.UUID
+		var dynArgs []json.RawMessage
+		var err error
+		for {
+			_, token, _, _, dynArgs, err = workers.RequestJobById(context.Background(), "", manifestJobID)
+			if err == jobqueue.ErrNotPending {
+				logrus.Infof("manifest job %v not pending, waiting for depsolve job to finish", manifestJobID)
+				time.Sleep(time.Second)
```
Comment (ondrejbudai) [inline internal/cloudapi/v2/v2.go:394]: 1 second is imho too much, let's be more optimistic. Previously, we used 50ms.

**Review by ondrejbudai — COMMENTED** [review]
This is awesome! It would be sweet if we can modify the `GET /composes/{id}` route to return depsolving errors but this can be done in a follow-up.

**Comment by ondrejbudai** [comment]
Needs a rebase.

**Review by ondrejbudai — APPROVED** [review]
Thanks!

**Review by teg — APPROVED** [review]

**Pull request merged**